### PR TITLE
Fix the bug for the strange color filled issue #24

### PR DESCRIPTION
**Pull request by dou-du**
The problem is caused by the chart.js library. When the data of the curve is not converging to zero, the filled color will not filled correctly. 

Here, I reuse the full range of the data. However, in order to get the scaling correctly. I gives the maximum and minimum number for the x range for the plot. 

The drawback of the fixed codes is that the scaling of the x axis not updating when you unchecking the PDOS. However, not scaling the y axis will let the users to feel the contribution of each PDOS.

### CODE REVIEW

File: widget_bandsplot/bandsplot.py [hunk 1]
```diff
@@ -112,7 +116,8 @@ def __init__(
                     )
                 )
 
-                temp_dos["dos"][i]["x"] = tx[index].tolist()
-                temp_dos["dos"][i]["y"] = ty[index].tolist()
+                ymin.append(min(ty[index]))
+                ymax.append(max(ty[index]))
 
+            self.dos_range = [min(ymin) * 1.05, max(ymax) * 1.05]
```
Comment (unkcpz) [inline widget_bandsplot/bandsplot.py:122]: I think this is the code that caused the drawback you described. Can you add a comment in the code so in the future we can understand it ?

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @dou-du, All good, just a minor request.

**Comment by unkcpz** [comment]
Hi @dou-du, since the other PR merged, please rebase this one with solving the confliction and request for review. Thanks!

File: widget_bandsplot/bandsplot.py (hunk 1 above)
Comment (dou-du) [reply widget_bandsplot/bandsplot.py:124]: Yes, this line is fixing the range of the density acoording to the maximum and miminum values. I added a comment as you suggested.

File: js/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     "@angular/material": "10.2.2",
     "@jupyter-widgets/base": "^1.1 || ^2 || ^3 || ^4",
     "bootstrap": "3.3.7",
-    "chart.js": "3.2.0",
+    "chart.js": "3.7.0",
```
Comment (unkcpz) [inline js/package.json:42]: I am pretty sure the version is then conflict with other two chart  library below. Can you check this file and make sure bandsplat js library can be installed by npm or yarn?

File: widget_bandsplot/bandsplot.py [hunk 3]
```diff
@@ -138,4 +142,8 @@ def __init__(
 
                 d_dos["backgroundColor"] = hex_alpha_to_rgba(bg_color, bg_alpha)
 
+            # Set the range of the axis of the density acoording to the
```
Comment (unkcpz) [inline widget_bandsplot/bandsplot.py:145]: 👌

File: widget_bandsplot/bandsplot.py [hunk 4]
```diff
@@ -109,8 +110,11 @@ def __init__(
             self.dos_fermienergy = dos["fermi_energy"]
             temp_dos = deepcopy(dos)
 
+            ymin = []
+            ymax = []
+
             # Convert data to the format for which chart.js can plot directly
-            for i, d_dos in enumerate(temp_dos["dos"]):
+            for _i, d_dos in enumerate(temp_dos["dos"]):
```
Comment (unkcpz) [inline widget_bandsplot/bandsplot.py:117]: If _i not used, we don't need call enumerate to get the index of list element.

File: js/lib/example.js [hunk 5]
```diff
@@ -112,10 +114,11 @@ var BandsplotView = widgets.DOMWidgetView.extend({
         var bands = this.model.get('bands');
         var fdos = this.model.get('dos');
         var yLimit = this.model.get('energy_range');
+        var dosRange = this.model.get('dos_range');
         var showFermi = this.model.get('plot_fermilevel');
         var showLegend = this.model.get('show_legend');
 
-        bandPlot(that.uuidCanvas, that.uuidTextbox, bands, fdos, showFermi, showLegend, yLimit, ['#000000', '#de2d26', '#444444']);
+        bandPlot(that.uuidCanvas, that.uuidTextbox, bands, fdos, showFermi, showLegend, yLimit, dosRange ['#000000', '#de2d26', '#444444']);
```
Comment (unkcpz) [inline js/lib/example.js:121]: I guess you miss a comma after `dosRange`? Is that possible to test it in voila?

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @dou-du, I think there is still things to improve, but very close.

File: js/lib/example.js (hunk 5 above)
Comment (dou-du) [reply js/lib/example.js]: Thanks for spoting this! I fixed it with the new commit.

File: widget_bandsplot/bandsplot.py (hunk 4 above)
Comment (dou-du) [reply widget_bandsplot/bandsplot.py]: Yes, I rmeoved it.

File: js/package.json (hunk 2 above)
Comment (dou-du) [reply js/package.json:42]: It seems working fine with v3.7.0. I did see any problem.

File: widget_bandsplot/bandsplot.py [hunk 6]
```diff
@@ -125,17 +129,21 @@ def __init__(
...
+                ymax.append(max(ty[index]))
 
                 # backgroundColor and backgroundAlpha combined to rgba value
                 bg_color = d_dos.get(
                     "backgroundColor", "#DCDCDC"
                 )  # grey for as default
 
                 bg_alpha = d_dos.get("backgroundAlpha", "0%")
-                bg_alpha = float(bg_alpha.strip("%") / 100)
+                bg_alpha = float(bg_alpha.strip("%")) / 100
```
Comment (unkcpz) [inline widget_bandsplot/bandsplot.py:141]: Nice catch.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Can you check the version of `chartjs-plugin-zoom` again? It is too low which conflict with `chartjs`.

File: js/package.json (hunk 2 above)
Comment (unkcpz) [reply js/package.json:42]: Thanks for checking, but if I go to the js folder I can not use yarn or npm to install. There is deps conflicting. My suggestion is we can fix that in another PR, would you mind to open an issue for that and investigate a bit on the annotation package? For this PR, I think it's all good then.

**Review by unkcpz — APPROVED** [review]
Thanks @dou-du

**Pull request merged**